ras_party/batch_process/batch_request.py

```diff
@@ -0,0 +1,87 @@
+import json
+import logging
+import structlog
+from flask import Blueprint, current_app, make_response, request
+from flask_httpauth import HTTPBasicAuth
+from werkzeug.exceptions import abort
+from ras_party.controllers import respondent_controller
+
+logger = structlog.wrap_logger(logging.getLogger(__name__))
+batch_request = Blueprint('batch_request', __name__)
+auth = HTTPBasicAuth()
+
+
+@batch_request.before_request
+@auth.login_required
+def before_respondent_view():
+    pass
+
+
+@auth.get_password
+def get_pw(username):
+    config_username = current_app.config['SECURITY_USER_NAME']
+    config_password = current_app.config['SECURITY_USER_PASSWORD']
+    if username == config_username:
+        return config_password
+
+
+@batch_request.route('/batch/respondents', methods=['DELETE'])
+def delete_user_data_marked_for_deletion():
+    """
+    Endpoint Exposed for Kubernetes Cronjob to delete all respondents and
+    its associated data marked for deletion
+    """
+    respondent_controller.delete_respondents_marked_for_deletion()
+    return '', 204
+
+
+@batch_request.route('/batch/requests', methods=['POST'])
+def batch():
```

Comment:
Is this particularly performant? It'll obviously cut down on network traffic and is very flexible, but for the example JSON its still N SQL transactions where it could just be one update transaction.

If we're likely to want to do a mix of different requests in a batch then it should stay like this. If we're only ever going to want to batch patch respondents, we should write an endpoint to do just that and optimise DB transactions for it.

Reply:
as discussed the reason of batch was to avoid multiple patch.